Source/BugsnagConfiguration.m

```diff
@@ -531,7 +532,7 @@ - (NSArray *)enabledReleaseStages {
 - (void)setEnabledReleaseStages:(NSArray *)newReleaseStages;
 {
     @synchronized (self) {
-        NSArray *releaseStagesCopy = [newReleaseStages copy];
+        NSSet<NSString *> *releaseStagesCopy = [newReleaseStages copy];
         _enabledReleaseStages = releaseStagesCopy;
         [self.config addMetadata:releaseStagesCopy
```

Comment:
Will this still be serialized when adding a set? I'm unsure off the top of my head whether BugsnagMetadata handles anything complex other than dictionaries + arrays.

Reply:
Good point.  I've reverted all `NSSet` use, and created an internal ticket to address the `NSSet` vs. `NSArray` issue more fully when time permits.

Reply:
I think it'd be preferable to use `NSSet` rather than `NSArray` on our public headers, as after the next major we won't be able to alter the type without making a breaking change. If it's going to take a long time to implement serialization of sets in metadata, then altering the implementation so that it continues to use `NSArray` under the hood seems good to me.

Reply:
Public API now presents `NSSet`.